src/java/com/unifina/data/ClockTick.java

```diff
@@ -0,0 +1,22 @@
+package com.unifina.data;
+
+import com.streamr.client.protocol.message_layer.ITimestamped;
+
+import java.util.Date;
+
+/**
+ * This is just a dummy event payload used to "tick" the clock in realtime and historical event queues.
+ */
+public class ClockTick implements ITimestamped {
+
+	private final Date timestamp;
+
+	public ClockTick(Date timestamp) {
+		this.timestamp = timestamp;
+	}
+
+	@Override
+	public Date getTimestampAsDate() {
+		return timestamp;
```

Comment:
Just realised that there is a similar problem with the constructor. Real fix would be to store timestamp as a long.
```
package com.unifina.data

import spock.lang.Specification

class ClockTickSpec extends Specification {
	void "does not expose internal representation"() {
		when:
		Date date = new Date(0)
		ClockTick tick = new ClockTick(date)
		tick.getTimestampAsDate().setTime(1)
		then:
		tick.getTimestampAsDate().getTime() == 0
	}
}
```
```suggestion
		return new Date(timestamp.getTime());
```